[bitnami/thanos] Add hash and time base partion for thanos store (#7049)
bitnami/charts: bitnami/thanos/README.md

In addition, you can also set an external ConfigMap with the configuration file. This is done by setting the `ruler.existingConfigmap` parameter. Note that this will override the previous option.

### Store time partions

Thanos store supports partition based on time.

> Setting time partitions will create n number of store deployment based on the number of items in the list. Each item must contain min and max time for querying in the supported format (see details here See details at https://thanos.io/tip/components/store.md/#time-based-partioning ).

Reviewer: This link is not correctly formatted:
```suggestion
Setting time partitions will create N number of store deployment based on the number of items in the list. Each item must contain min and max time for querying in the supported format (see details at [Thanos documentation](https://thanos.io/tip/components/store.md/#time-based-partioning)).
```

> Setting time partitions will create N number of store deployment based on the number of items in the list. Each item must contain min and max time for querying in the supported format (see details at [Thanos documentation](https://thanos.io/tip/components/store.md/#time-based-partitioning)).

Reviewer: ```suggestion
Setting time partitions will create N number of store statefulsets based on the number of items in the `timePartitioning` list. Each item must contain the min and max time for querying in the supported format (find more details at [Thanos documentation](https://thanos.io/tip/components/store.md/#time-based-partitioning)).
```

> Leaving this empty list ([]) will create a single store for all data.

Reviewer: ```suggestion
> Note: leaving this empty list ([]) will create a single store for all data.
```

> > Note: leaving this empty list ([]) will create a single store for all data.

Reviewer: ```suggestion
> Note: leaving the `timePartitioning` list empty (`[]`) will create a single store for all data.
```

> Example - This will create 3 stores:

Reviewer: ```suggestion
For instance, to use 3 stores you can use a **values.yaml** like the one below:
```
